spec/unit/client_spec.rb

```diff
@@ -235,10 +235,20 @@
     include_context 'shared context'
 
     it "calls the correct resource's get_all method" do
-      expect(OneviewSDK::ServerProfile).to receive(:get_all).with(@client)
+      expect(OneviewSDK::API200::ServerProfile).to receive(:get_all).with(@client)
       @client.get_all('ServerProfiles')
     end
 
+    it 'accepts API version and variant parameters' do
```

Comment:
Suggestion: It may be interesting to have a test for exception. But the code looks good for me.